src/integration_tests/mission_takeoff_land.cpp

```diff
@@ -0,0 +1,168 @@
+    auto action = std::make_shared<Action>(system);
+
+    PathChecker pc;
+
+    telemetry->subscribe_position([&pc](Telemetry::Position position) {
+        pc.check_current_alitude(position.relative_altitude_m);
+    });
+
+    std::vector<Telemetry::LandedState> landed_states_template = {
+        Telemetry::LandedState::OnGround,
+        Telemetry::LandedState::TakingOff,
+        Telemetry::LandedState::InAir,
+        Telemetry::LandedState::Landing,
+        Telemetry::LandedState::OnGround,
+        Telemetry::LandedState::TakingOff,
+        Telemetry::LandedState::InAir,
+        Telemetry::LandedState::Landing,
+        Telemetry::LandedState::OnGround};
+
+    std::vector<Telemetry::LandedState> landed_states;
+    Telemetry::LandedState landed_state_previous = Telemetry::LandedState::Unknown;
+    telemetry->subscribe_landed_state(
+        [&landed_states, &landed_state_previous](Telemetry::LandedState landed_state) {
+            if (landed_state_previous != landed_state) {
+                LogInfo() << landed_state;
+                landed_state_previous = landed_state;
+                landed_states.push_back(landed_state);
+            }
+        });
+
+    while (!telemetry->health_all_ok()) {
+        LogInfo() << "Waiting for system to be ready";
+        LogDebug() << "Health: " << telemetry->health();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    LogInfo() << "System ready";
+    LogInfo() << "Creating and uploading mission";
+
+    Mission::MissionItem new_item{};
+    Mission::MissionPlan mission_plan{};
+
+    new_item.latitude_deg = 47.398170327054473;
+    new_item.longitude_deg = 8.5456490218639658;
+    new_item.relative_altitude_m = mission_altitude_m;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Takeoff;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398370327054473;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Land;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398570327054473;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Takeoff;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398770327054473;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Land;
+    mission_plan.mission_items.push_back(new_item);
+
+    {
+        LogInfo() << "Uploading mission...";
+        auto prom = std::make_shared<std::promise<void>>();
+        auto future_result = prom->get_future();
+        mission->upload_mission_async(mission_plan, [prom](Mission::Result result) {
+            ASSERT_EQ(result, Mission::Result::Success);
+            prom->set_value();
+            LogInfo() << "Mission uploaded.";
+        });
+
+        auto status = future_result.wait_for(std::chrono::seconds(2));
+        ASSERT_EQ(status, std::future_status::ready);
+        future_result.get();
+    }
+
```

Comment:
Can you also download the mission again here and check if the items match again, so that we know the inverse works?

Reply:
Don't understand what exactly do you mean. When whole mission is generated and uploaded into vehicle it is also possible to download this mission?

Reply:
Basically I mean this: https://github.com/mavlink/MAVSDK/blob/ff986d803be36de0f2bed850c3adb4ade98a95a1/src/integration_tests/mission.cpp#L196-L220

Reply:
I have added it in 00fe5786f2e3e798e2ba183bdf6c4e7c9fddbcd0